maggy/core/environment/base.py

```diff
@@ -0,0 +1,170 @@
+
+
+    def mkdir(self, hdfs_path):
+        pass
+
+    def isdir(self, dir_path, project=None):
+        return os.path.exists(dir_path)
+
+    def ls(self, dir_path):
+        return os.listdir(dir_path)
+
+    def delete(self, path):
+
+        if self.exists(path):
+            if os.path.isdir(path):
+                os.rmdir(path)
+            elif os.path.isfile(path):
+                os.remove(path)
+
+    def dump(self, data, hdfs_path):
+        head_tail = os.path.split(hdfs_path)
+        if not os.path.exists(head_tail[0]):
+            os.mkdir(head_tail[0])
+        file = self.open_file(hdfs_path, flags='w')
+        file.write(data)
+
+    def get_ip_address(self):
+        sc = util.find_spark().sparkContext
+        return sc._conf.get("spark.driver.host")
+
+    def get_constants(self):
+        pass
+
+    def open_file(self, hdfs_path, flags='r', buff_size=-1):
+        return open(hdfs_path, mode=flags, buffering=buff_size)
+
+    def get_training_dataset_path(self, training_dataset, featurestore=None, training_dataset_version=1):
+        pass
+
+    def get_training_dataset_tf_record_schema(self, training_dataset, training_dataset_version=1, featurestore=None):
+        pass
+
+    def get_featurestore_metadata(self, featurestore=None, update_cache=False):
+        pass
+
+    def init_ml_tracking(self, app_id, run_id):
+        pass
+
+    def log_searchspace(self, app_id, run_id, searchspace):
+        pass
+
+    def connect_host(self,server_sock, server_host_port, exp_driver):
+        if not server_host_port:
+            server_sock.bind(("", 0))
+            host = self.get_ip_address()
+            port = server_sock.getsockname()[1]
+            server_host_port = (host, port)
+
+        else:
+            server_sock.bind(server_host_port)
+
+        server_sock.listen(10)
+
+        return server_sock, server_host_port
+
+    def _upload_file_output(self, retval, hdfs_exec_logdir):
+        pass
+
+    def project_path(self):
+        return os.getcwd()
+
+    def get_user(self):
+        return ""
+
+    def project_name(self):
+        return ""
+
+    def finalize_experiment(self,
+            experiment_json,
+            metric,
+            app_id,
+            run_id,
+            state,
+            duration,
+            logdir,
+            best_logdir,
+            optimization_key
+                            ):
+        pass
+
+    def str_or_byte(self, str):
+        return str
+
+    def get_executors(self, sc):
+        try:
+            if sc._conf.get("spark.dynamicAllocation.enabled") == "true":
+                maxExecutors = int(sc._conf.get("spark.dynamicAllocation.maxExecutors"))
+            else:
+                maxExecutors = int(sc._conf.get("spark.executor.instances"))
+
+            return maxExecutors
+        except:  # noqa: E722
+            raise RuntimeError(
+                "Failed to find some of the spark.databricks properties."
+            )
```

Comment:
Same as in the databricks env, split it up into two try/except so you are able to tell the user which properties are missing. Also if they are going to open an issue with the error message, it will be easier for us.